src/main/scala/temple/DSL/semantics/Validator.scala

```diff
@@ -112,9 +112,33 @@ private class Validator private (templefile: Templefile) {
     renameBlock(structName, struct)
 
     val newAttributes = validateAttributes(struct, context)
-    validateMetadata(struct.metadata, context)
+    val newMetadata   = validateStructMetadata(struct.metadata, newAttributes, context)
 
-    StructBlock(newAttributes, struct.metadata)
+    StructBlock(newAttributes, newMetadata)
+  }
+
+  private def validateStructMetadata[M >: Metadata.StructMetadata <: Metadata](
+    metadata: Seq[M],
+    attributes: Map[String, AbstractAttribute],
+    context: SemanticContext,
+  ): Seq[M] = {
+    validateMetadata(metadata, context)
+
+    val noUpdates = attributes.valuesIterator.forall(attributes =>
+      attributes.accessAnnotation.contains(Annotation.Server)
+      || attributes.accessAnnotation.contains(Annotation.ServerSet),
+    )
+    if (noUpdates) {
+      var foundOmit = false
+      val newMetadata = metadata map {
+          case Metadata.Omit(endpoints) =>
+            foundOmit = true
+            Metadata.Omit(endpoints + Metadata.Endpoint.Update)
+          case m => m
+        }
+      if (foundOmit) newMetadata
+      else newMetadata :+ Metadata.Omit(Set(Metadata.Endpoint.Update))
+    } else metadata
```

Comment:
This whole function seems a little odd in its control follow, anyone got a better suggestion of a way to lay it out?

Reply:
See what you mean, could you add a comment to explain why you're doing what you're doing